### PR TITLE
edits for DFE-alpha

### PR DESCRIPTION
**Pull request by silastittes**

### CODE REVIEW

File: workflows/ts2fs.py [hunk 1]
```diff
@@ -2,6 +2,58 @@
...
+        for mtypes in dfe['mutation_types']:
+            if mtypes['is_neutral']:
+                annot_intervals = np.array(dfe['intervals'])
+                #assumes netural is first proportion in list
+                proportion = dfe['proportions'][0]
+                ratemap = mask_to_ratemap(annot_intervals, sequence_length, mutation_rate, proportion)
+                slim_simulation_time = ts.metadata['SLiM']['tick']
+                for t in ts.trees():
+                    if t.num_edges > 0:
+                        exp_neutral_subs += max(0, (slim_simulation_time - t.root) *
```
[the remark this answers is not in the document]
Comment (nspope) [reply workflows/ts2fs.py:51]: What's the context-- do you want to actually add mutations above the root (so would need a discrete number)? Or just calculate ratio of neutral vs deleterious subs (in which case I guess you're thinking to add some Poisson noise, to make the comparison more realistic)?

[the remark this answers is not in the document]
Comment (andrewkern) [reply workflows/ts2fs.py:51]: this is input for `DFE-alpha`. We want the discrete number, but can round the expected value.

[the remark this answers is not in the document]
Comment (nspope) [reply workflows/ts2fs.py:51]: Either should be fine-- but the simulation end point is arbitrary, and recapitation will extend trees beyond it in parts of the sequence (so the intervals where substitutions can accumulate are also arbitrary). 

What is the analogue with real datasets? Substitutions on a long branch separating an outgroup (in which case I guess the "recapitated" parts mimic ILS)? Is that what DFE-alpha wants?

File: workflows/ts2fs.py [hunk 2]
```diff
@@ -288,7 +363,9 @@ def _generate_grapes_fs(neu_fs, nonneu_fs, output, is_folded, seq_len, sample_si
         nonneu_prop float: Proportion of non-neutral mutations
     """
     neu_len = round(seq_len * kwargs['neu_prop'])
+    neu_fs[0] = neu_len - sum(neu_fs[1:]) 
     nonneu_len = round(seq_len * kwargs['nonneu_prop'])
+    nonneu_fs[0] = nonneu_len - sum(nonneu_fs[1:])
```
Comment (silastittes) [inline workflows/ts2fs.py:369]: Need to add something similar in `_generate_dfe_alpha_fs`. Right now the first bin is ALL positions including mutated and substitutions. This is the right format form some software, but not for DFE-alpha.

File: workflows/ts2fs.py [hunk 3]
```diff
@@ -45,9 +104,16 @@ def f(x):
             ts = ts.keep_intervals(annot_intervals)
             exon_len = np.sum(annot_intervals[:,1]-annot_intervals[:,0])
             seq_len += exon_len
+            non_neutral_anc_count += exon_len * nonneu_prop
+            neutral_anc_count += exon_len * neutral_prop
         else:
-            contig = species.get_contig(chrm)
-            seq_len += contig.recombination_map.sequence_length
+            seq_len += chrom_length
```
Comment (silastittes) [inline workflows/ts2fs.py:110]: We should subtract mask length here.

File: workflows/ts2fs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewkern) [reply workflows/ts2fs.py:51]: yep. it defines a SFS that goes from x =  {X_0, X_1, ..., X_N) where X_0 is the number of monomorphic, ancestral sites and X_N is the number of fixed, derived sites

[the remark this answers is not in the document]
Comment (nspope) [reply workflows/ts2fs.py:51]: I see ... so the parts of the sequence that are recapitated (don't have subs) are mimicking intervals where there's ILS. This all makes sense to me. Sampling Poisson # of neutral subs seems like the most natural thing, to me.

[the remark this answers is not in the document]
Comment (silastittes) [reply workflows/ts2fs.py:51]: poisson draw added. thanks!

File: workflows/ts2fs.py [hunk 4]
```diff
@@ -85,6 +151,22 @@ def f(x):
...
+    
+    #add in ancestral counts
+    mut_afs["neutral"][0] = neutral_anc_count
+    mut_afs["non_neutral"][0] = non_neutral_anc_count
+    
+    # Remove mutated sites from first bin
+    mut_afs["neutral"][0] = mut_afs["neutral"][0] - sum(mut_afs["neutral"][1:])
+    mut_afs["non_neutral"][0] = mut_afs["non_neutral"][0] - sum(mut_afs["non_neutral"][1:])
+
+    print(f"AFSs\nNeutral: {mut_afs['neutral']}\nNon-neutral: {mut_afs['non_neutral']}")
```
Comment (silastittes) [inline workflows/ts2fs.py:169]: poisson draw and removing mutations from ancestral monomorphic counts

**Comment by silastittes** [comment]
closing because now subset of what's in #120

**Comment by silastittes** [comment]
This one has some edits we need. Should be good to merge

**Pull request closed without merging**